### PR TITLE
Upgrade gradle-baseline

### PR DESCRIPTION
**Pull request by schlosna**
**Goals (and why)**:
AtlasDB's gradle-baseline version is significantly out of date and is not being kept up-to-date by excavators. Similar to #5043 , this upgrades the gradle-baseline version used so that atlasdb is enforcing (most of) the checks baseline provides

**Implementation Description (bullets)**:

Steps below can be seen in individual commits on this PR

- Apply gradle-palantir-java-format plugin
- Bump gradle-baseline plugin version
- Update baseline configs
- Bump error-prone and immutables versions for compatibility with error-prone
- Apply automated gradle-baseline refactorings (`./gradlew compileJava compileTestJava -PrefasterApply -PerrorProneApply`)
- Reformat (`./gradlew format`)
- Temporarily relax some error-prone checks from error to warning (to be fixed iteratively in follow up PRs after merge)
- Workaround error-prone fatal error with lambda method reference check
- Unify checkstyle configuration to gradle-baseline version
- Standardized IntelliJ Idea configuration to gradle-baseline version
- Suppress some checkstyle rules until they can be remediated 
- Fix several missing Apache license headers
- Upgrade to gradle 6.7
- Standardize baseline configuration in build.gradle

**Testing (What was existing testing like?  What have you done to improve it?)**:
Local compilation

**Concerns (what feedback would you like?)**:
The reformat generated a ton of diffs, we may want to split that out separately, but I wanted to use the automated refactorings which also generate a bunch of diffs.

**Where should we start reviewing?**:
`build.gradle` has most of the changes and temporary exclusions.

**Priority (whenever / two weeks / yesterday)**:
Whenever

### CODE REVIEW

File: .baseline/checkstyle/checkstyle-suppressions.xml
Comment (schlosna) [inline .baseline/checkstyle/checkstyle-suppressions.xml:36]: moved to custom-suppressions.xml per gradle-baseline standard

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/TableSplittingKeyValueService.java
Comment (carterkozak) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/TableSplittingKeyValueService.java:381]: nice

File: .baseline/checkstyle/custom-suppressions.xml
Comment (schlosna) [inline .baseline/checkstyle/custom-suppressions.xml:29]: I opted to skip these checks for now so that we have the vast majority of improved baseline checks, and we can burn these down in smaller, independent PRs that are easier to review

File: build.gradle
Comment (schlosna) [inline build.gradle:27]: sorted these to simplify future updates.

File: changelog/@unreleased/pr-5044.v2.yml
Comment (carterkozak) [inline changelog/@unreleased/pr-5044.v2.yml:34]: I don't think we need a changelog for these unless there are specific fixes we want to call out.

If we keep the changelog we should remove the template components and write a a short description of the pieces consumers may be interested in.

File: build.gradle
Comment (schlosna) [inline build.gradle:39]: applied by default now https://github.com/palantir/gradle-baseline/blob/32b092e918cb3f46380273fd7957a26b09ee0d6a/gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/Baseline.java#L36

File: build.gradle
Comment (schlosna) [inline build.gradle:38]: applied by default now https://github.com/palantir/gradle-baseline/blob/32b092e918cb3f46380273fd7957a26b09ee0d6a/gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/Baseline.java#L35

File: build.gradle
Comment (schlosna) [inline build.gradle:68]: use standard formatting to streamline the baseline refactorings

File: build.gradle
Comment (schlosna) [inline build.gradle:142]: relaxing from error to warning for now, will burn down in separate PRs

File: gradle/baseline.gradle
Comment (schlosna) [inline gradle/baseline.gradle:4]: moved all of this to `build.gradle` and this is no longer opt-in, its default on everywhere

File: gradle/idea.gradle
Comment (schlosna) [inline gradle/idea.gradle:14]: defaulting to gradle-baseline's IntelliJ Idea integration configs. Checkstyle now works properly and compiles fine for me in IDEA 2020.3 CE EAP.

File: gradle/shared.gradle
Comment (schlosna) [inline gradle/shared.gradle:30]: We can probably clean up more of these in a follow up PR

File: gradle/versions.gradle
Comment (schlosna) [inline gradle/versions.gradle:31]: AtlasDB has had JDK8 as minimum dependency for a while, and this old version of checkstyle isn't compatible with gradle-baseline's checkstyle rules, and this old version would breaks Idea Checkstyle integration, but now this all works properly.

File: versions.props
Comment (schlosna) [inline versions.props:48]: bumping this was necessary to fix gradle-baseline & error-prone compatibility

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/TableSplittingKeyValueService.java
Comment (schlosna) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/TableSplittingKeyValueService.java:381]: This was an automated gradle-baseline refactor

**Comment by carterkozak** [comment]
I think the CI failures are related to the scale of error-prone compiler warnings, they seem to go away when there are fewer failures.

**Comment by schlosna** [comment]
@carterkozak should we go the route of https://github.com/palantir/atlasdb/pull/5048 to disable those specific checks if they're causing overhead and CI flakes?

**Comment by carterkozak** [comment]
I've updated that PR to set most of them to WARN, disable a couple, and fix the noisy ones with safe automatic fixes. It seems to be passing.

**Comment by carterkozak** [comment]
It targets this PR, so we can pull in any subset of the changes we want

File: gradle/baseline.gradle
Comment (jkozlowski) [reply gradle/baseline.gradle:4]: So reading through it looks like this was doing some per main/test package differences, but I think it's fine. We should quickly drive all the turned off warnings.

**Comment by jkozlowski** [comment]
Ok, so I went through the chain of commits, and individually they make sense. I scrolled through the large automated format, and read through errorprone auto-refactorings.

I think this makes sense and needs to be merged quickly, before it gets stale (it already started accumulating merge conflicts).

@schlosna are we now running with the equivalent of -WError for all of compiler|checkstyle|errorprone?

**Review by jkozlowski — APPROVED** [review]

File: .baseline/idea/intellij-java-palantir-style.xml
Comment (jeremyk-91) [inline .baseline/idea/intellij-java-palantir-style.xml:117]: 💯

File: atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/AtlasConsoleBinder.java
Comment (jeremyk-91) [inline atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/AtlasConsoleBinder.java:103]: Is this OK? Lines 114-115 suggests that `(authorized)` should be indicated on both lines

**Review by jeremyk-91 — COMMENTED** [review]
Thanks a lot for doing this! I think there's one thing to validate that I'm not sure is correct, since I remember this was one of the initial sticking points we had when refaster was first going out. @carterkozak I'm tagging you as I think you might know as well, if this is safe?

I put a "do not merge" just so this doesn't accidentally merge without us making a decision on this: once we get confirmation we should be fine to proceed.

-Werror question: given description I'd say probably not (since things were moved to warn to avoid erroring)?

File: atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/AtlasConsoleBinder.java
Comment (jeremyk-91) [reply atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/AtlasConsoleBinder.java:103]: Had a look internally - this is probably OK (see e.g. the large internal product's class that inspects atlas's lock service)

**Review by jeremyk-91 — APPROVED** [review]
validated my earlier concern. Thanks a lot for doing this!

**Comment by jkozlowski** [comment]
ok so @schlosna is it possible to move the ones we have disabled to just be suppressed and then -Werror everything? that's certainly how we run atlas-proxy internally

**Comment by schlosna** [comment]
@jkozlowski right now this PR just upgrades atlasdb to build against a modern version of gradle-baseline so that we can keep baseline up to date with excavators. We will need separate PRs to fix up the remaining checks we had to reduce from error to warning to treat those as errors, and then we'll also need to promote warnings to error (equivalent to `-Wall -Werror`). As you've seen, this PR is already too large and quickly accrues conflicts, so would prefer we land this first then work those down so we're at least progressing in the right direction.

**Comment by schlosna** [comment]
After this goes green and merges, I think we can bump https://errorprone.info/bugpattern/BoxedPrimitiveEquality and https://errorprone.info/bugpattern/ReferenceEquality to error and fix the findings.

**Pull request merged**
